### DISCUSSION TITLE
Linking external images securely - Azure Blob

### DISCUSSION THREAD

**Discussion by petrsx** [grafana/grafana#56585] [Alerting]
Hi, we would like to send Images to MS Teams using External image storage - Azure Storage Account. However only public Containers are supported.
This is a bit a issue as we need to be very careful not to expose any information (like IP Address or Resource Endpoint).

I guess getting SAS token as part of the Alerting would be time consuming change for you guys. However i have one idea which may work and be simple.

Below is Azure Blob config:

```
#################################### External image storage ##########################
[external_image_storage]
...

[external_image_storage.azure_blob]
;account_name =
;account_key =
;container_name =
```

Would be possible to add `sas_token` variable, which will be attached to the image URL as parameter like on the screenshot below?

![image](https://user-images.githubusercontent.com/18548253/194663891-cada11d3-2223-4ca6-8a93-4f447ff2eaba.png)
https://learn.microsoft.com/en-us/azure/storage/common/storage-sas-overview#sas-token

The idea is to change the Blob Container to Private and manually (or by any custom made automation) generate SAS token for the Container and attach it to the URL which is then sent as link to the user.

It will be not super secure, however it is possibile to set expiration of the SAS token and rotate it as ENV variable in our scheduled deployment pipeline.

So i guess this small change can be big improvement in term of data security.

Please let me know if you have any question or comments.
Thank you

Edit:
Already opened PR: https://github.com/grafana/grafana/pull/56598

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
